pkg/config/config.go

```diff
@@ -376,6 +379,35 @@ func Load() error {
 		return err
 	}
 
+	// We have to init the secrets package before we can use it to decrypt
+	// anything.
+	secrets.Init(
+		Datadog.GetString("secret_backend_command"),
+		Datadog.GetStringSlice("secret_backend_arguments"),
+		Datadog.GetInt("secret_backend_timeout"),
+		Datadog.GetInt("secret_backend_output_max_size"),
+	)
+
+	if Datadog.IsSet("secret_backend_command") {
+		// Viper doesn't expose the final location of the file it
+		// loads. Since we are searching for 'datadog.yaml' in multiple
+		// localtions we let viper determine the one to use before
+		// updating it.
+		conf, err := yaml.Marshal(Datadog.AllSettings())
+		if err != nil {
+			return fmt.Errorf("unable to marshal configuration to YAML to decrypt secrets: %v", err)
+		}
+
+		finalConfig, err := secrets.Decrypt(conf)
+		if err != nil {
+			return fmt.Errorf("unable to decrypt secret from datadog.yaml: %v", err)
+		}
+		r := bytes.NewReader(finalConfig)
+		if err = Datadog.MergeConfig(r); err != nil {
```

Comment:
So just a note for posterity, the `MergeConfig()` will merge the new _"source"_ (`finalConfig`) into the _"target"_ (Datadog), so new values will override old ones.